From 77a4011ce7b2c69faacf666bbaec09336894d7db Mon Sep 17 00:00:00 2001
From: Rafael Mardojai CM <mardojai.cardenas@gmail.com>
Date: Wed, 29 Sep 2021 20:15:50 -0500
Subject: [PATCH] Separate trying to set locale and gettext domains Fix Blanked
 not launching if locale.bindtextdomain and locale.textdomain don't exist

Co-authored-by: mio <miyopan@e.email>
---
 blanket/blanket.in | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/blanket/blanket.in b/blanket/blanket.in
index 7d6d3ec..4f84102 100755
--- a/blanket/blanket.in
+++ b/blanket/blanket.in
@@ -18,10 +18,16 @@ sys.path.insert(1, pkgdatadir)
 signal.signal(signal.SIGINT, signal.SIG_DFL)
 gettext.install('blanket', localedir)
 
-locale.bindtextdomain('blanket', localedir)
-locale.textdomain('blanket')
-gettext.bindtextdomain('blanket', localedir)
-gettext.textdomain('blanket')
+try:
+  locale.bindtextdomain('blanket', localedir)
+  locale.textdomain('blanket')
+except:
+  print('Cannot set locale.')
+try:
+  gettext.bindtextdomain('blanket', localedir)
+  gettext.textdomain('blanket')
+except:
+  print('Cannot load translations.')
 
 if __name__ == '__main__':
     import gi
